app/shared/video-player/common/ui/progress/MediaProgressSlider.kt

```diff
@@ -219,6 +244,77 @@ fun MediaProgressSlider(
                 )
             }
         }
+
+        var mousePosX by rememberSaveable { mutableStateOf(0f) }
+        var thumbWidth by rememberSaveable { mutableIntStateOf(0) }
+        var sliderWidth by rememberSaveable { mutableIntStateOf(0) }
+        
+        val previewTimeText by remember {
+            derivedStateOf {
+                val percent = mousePosX.minus(thumbWidth / 2).div(sliderWidth - thumbWidth)
+                    .coerceIn(0f, 1f)
+                val previewTimeMillis = state.totalDurationMillis.times(percent).toLong()
+                renderSeconds(previewTimeMillis / 1000, state.totalDurationMillis / 1000).substringBefore(" ")
+            }
+        }
+        val hoverInteraction = remember { MutableInteractionSource() }
+        var isHoveredAsState by hoverInteraction.collectIsHoveredAsState() as MutableState 
```

Comment:
`as MutableState` 是不好的

可以考虑改成 
```suggestion
        val isHovered by hoverInteraction.collectIsHoveredAsState() // works only for desktop
        var isPressed by remember { mutableStateOf(false) }
        val showPreviewTime by remember {
            derivedStateOf {
                isHovered || isPressed
            }
        }
```

然后下面的 if 使用 `showPreviewTime`

Reply:
没有使用 `rememberSaveable` 因为我觉得 configuration change (例如系统自动切换为深色模式) 的时候应该用户的手势也应当取消, 或者说 Compose 会重新 emit 事件.